extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -366,29 +380,33 @@ public long getLikeCount() throws ParsingException {
     public long getDislikeCount() throws ParsingException {
         assertPageFetched();
 
-        String dislikesString = "";
+        String dislikesString = null;
         try {
-            try {
-                dislikesString = getVideoPrimaryInfoRenderer().getObject("sentimentBar")
-                        .getObject("sentimentBarRenderer").getString("tooltip").split("/")[1];
-            } catch (final NullPointerException e) {
-                // If this kicks in our button has no content and therefore ratings must be disabled
-                if (playerResponse.getObject("videoDetails").getBoolean("allowRatings")) {
-                    throw new ParsingException(
-                            "Ratings are enabled even though the dislike button is missing", e);
+            dislikesString = getVideoPrimaryInfoRenderer().getObject("sentimentBar")
+                    .getObject("sentimentBarRenderer").getString("tooltip");
+            if (dislikesString != null && dislikesString.contains("/")) {
+                dislikesString = dislikesString.split("/")[1];
+            } else {
+                // Calculate dislike with average rating and like count
+                long likes = getLikeCount();
+                double averageRating = playerResponse.getObject("videoDetails").getDouble("averageRating");
+
+                if (likes != -1 && averageRating > 1) {
+                    // If averageRating can't be gathered, it will be 0,
+                    // but we also can't divide by 0 so we need > 1
+                    return Math.round(likes * ((5 - averageRating) / (averageRating - 1)));
                 }
-                return -1;
             }
-            return Integer.parseInt(Utils.removeNonDigitCharacters(dislikesString));
-        } catch (final NumberFormatException nfe) {
-            throw new ParsingException("Could not parse \"" + dislikesString + "\" as an Integer",
-                    nfe);
-        } catch (final Exception e) {
-            if (getAgeLimit() == NO_AGE_LIMIT) {
-                throw new ParsingException("Could not get dislike count", e);
+
+            if (dislikesString != null) {
+                return Integer.parseInt(Utils.removeNonDigitCharacters(dislikesString));
             }
-            return -1;
+
+        } catch (final Exception e) {
```

Comment:
Java allows not to catch these exception but we want to:
- `Integer.parseInt()` can throw a `NumberFormatException`.
- `getLikeCount()` can throw a `ParsingException`.

We don’t want dislike to fail and show a snackbar if YouTube changes something about dislike (that could be soon).

Reply:
> Java allows not to catch these exception but we want to.

*Confused*
It does allow catching of ``RuntimeException``s

> We don’t want dislike to fail and show a snackbar if YouTube changes something about dislike (that could be soon).

I don't understand that completely. YT can always change anything, but this is no reason to have an empty try-catch-block.
I think there should be an error message if an error occurs when the dislikes are parsed. If not then why is there a declaration for ``throws ParsingException`` in the first place?

So what I would do:
* Remove the existing try-catch block
* ``NumberFormatException`` from ``return Integer.parseInt(Utils.removeNonDigitCharacters(dislikesString));`` can be catched with a try-catch block and then be re-thrown as a ``new ParsingException("Parsing of dislikes failed", ex)``.

Reply:
@B0pol 
Ping

Reply:
Yes it can always change but specifically there it is known it will may be removed and there won't be ways to gather it.

Usually it's changes but the feature is still there so we want users to report it. For dislikes we neither want users to report it nor annoying snackbar until next update